### PR TITLE
Fix: Allow Issue Matching across org/repos

### LINKED ISSUE #60: Issue Matching Fails, with invalid payload object.
The issue matchmaking (issue contributor recommendation) functionality should work across multiple orgs/repos and not be restricted within single repo.

### PR DESCRIPTION
**Pull request by shiv810**
Resolves #60 

- Allows Issue Matching outside repos and orgs. 
- New pg func for handling the issue matching

### CODE REVIEW

File: src/handlers/issue-matching.ts [hunk 1]
```diff
@@ -91,10 +91,14 @@ export async function issueMatching(context: Context<"issues.opened" | "issues.e
         return null;
       }
     });
-    const issueList = (await Promise.all(fetchPromises)).filter((issue) => issue !== null);
+    const issueList = await Promise.allSettled(fetchPromises);
```
Comment (0x4007) [inline src/handlers/issue-matching.ts:94]: Isn't this error prone if a single request fails then everything fails?

File: supabase/config.toml [hunk 2]
```diff
@@ -141,7 +141,7 @@ redirect_uri = ""
 url = ""
 
 [analytics]
-enabled = true
+enabled = false
```
Comment (0x4007) [inline supabase/config.toml:144]: Why no analytics

Comment (sshivaditya) [reply supabase/config.toml]: Fixed.

File: src/adapters/supabase/helpers/issues.ts [hunk 3]
```diff
@@ -193,4 +193,34 @@ export class Issue extends SuperSupabase {
...
+
+  async findSimilarIssuesToMatch({ markdown, currentId, threshold }: FindSimilarIssuesParams): Promise<IssueSimilaritySearchResult[] | null> {
+    // Create a new issue embedding
+    try {
+      const embedding = await this.context.adapters.voyage.embedding.createEmbedding(markdown);
+      const { data, error } = await this.supabase.rpc("find_similar_issues_to_match", {
+        current_id: currentId,
+        query_embedding: embedding,
+        threshold,
+        top_k: 5,
```
Comment (0x4007) [inline src/adapters/supabase/helpers/issues.ts:205]: Why 5?

File: tests/__mocks__/adapter.ts [hunk 4]
```diff
@@ -114,6 +114,14 @@ export function createMockAdapters(context: Context) {
 
           return [];
         }),
+        findSimilarIssuesToMatch: jest.fn(async (params: { markdown: string; threshold: number; currentId: string }) => {
+          if (params.currentId === "task_complete") {
+            return [{ id: "similar3", similarity: 0.98 }];
```
Comment (0x4007) [inline tests/__mocks__/adapter.ts:119]: Why similar3

**Review by 0x4007 — APPROVED** [review]

File: src/adapters/supabase/helpers/issues.ts (hunk 3 above)
Comment (0x4007) [reply src/adapters/supabase/helpers/issues.ts:205]: Probably not but was just curious why you chose 5 but based on brief research it seems appropriate for our usecase.

**Pull request merged**